File: sivemor-app/src/main/java/mx/edu/utez/sivemorapp/modules/evaluaciones/dtos/EvaluacionRequestDTO.java
package mx.edu.utez.sivemorapp.modules.evaluaciones.dtos;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import mx.edu.utez.sivemorapp.kernel.enums.*;

import java.math.BigDecimal;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class EvaluacionRequestDTO {
    private Long id;
    private Long idVerificacion;
    private Long idTecnico;

    private EstadoLuces lucesGalibo;
    private EstadoLuces lucesAltas;
    private EstadoLuces lucesBajas;
    private EstadoLuces lucesDemarcadorasDelanteras;
    private EstadoLuces lucesDemarcadorasTraseras;
    private LucesIndicadoras lucesIndicadoras;
    private EstadoFaros faroIzquierdo;
    private EstadoFaros faroDerecho;
    private EstadoLuces lucesDireccionalesDelanteras;
    private EstadoLuces lucesDireccionalesTraseras;

    private EstadoRines llantasRinesDelanteros;
    private EstadoRines llantasRinesTraseros;
    private EstadoMasas llantasMasasDelanteras;
    private EstadoMasas llantasMasasTraseras;

    private BigDecimal llantasPresionDelanteraIzquierda;
    private BigDecimal llantasPresionDelanteraDerecha;
    private BigDecimal llantasPresionTraseraIzquierda1;
    private BigDecimal llantasPresionTraseraIzquierda2;
    private BigDecimal llantasPresionTraseraDerecha1;
    private BigDecimal llantasPresionTraseraDerecha2;

    private BigDecimal llantasProfundidadDelanteraIzquierda;
    private BigDecimal llantasProfundidadDelanteraDerecha;
    private BigDecimal llantasProfundidadTraseraIzquierda1;
    private BigDecimal llantasProfundidadTraseraIzquierda2;
    private BigDecimal llantasProfundidadTraseraDerecha1;
    private BigDecimal llantasProfundidadTraseraDerecha2;

    private EstadoBirlos llantasBirlosDelanteraIzquierda;
    private EstadoBirlos llantasBirlosDelanteraDerecha;
    private EstadoBirlos llantasBirlosTraseraIzquierda;
    private EstadoBirlos llantasBirlosTraseraDerecha;

    @Builder.Default
    private Integer llantasBirlosDelanteraIzquierdaNum = 0;
    @Builder.Default
    private Integer llantasBirlosDelanteraDerechaNum = 0;
    @Builder.Default
    private Integer llantasBirlosTraseraIzquierdaNum = 0;
    @Builder.Default
    private Integer llantasBirlosTraseraDerechaNum = 0;

    private EstadoTuercas llantasTuercasDelanteraIzquierda;
    private EstadoTuercas llantasTuercasDelanteraDerecha;
    private EstadoTuercas llantasTuercasTraseraIzquierda;
    private EstadoTuercas llantasTuercasTraseraDerecha;

    @Builder.Default
    private Integer llantasTuercasDelanteraIzquierdaNum = 0;
    @Builder.Default
    private Integer llantasTuercasDelanteraDerechaNum = 0;
    @Builder.Default
    private Integer llantasTuercasTraseraIzquierdaNum = 0;
    @Builder.Default
    private Integer llantasTuercasTraseraDerechaNum = 0;

    private EstadoBrazoPitman brazoPitman;
    private EstadoManijas manijasDePuertas;
    private EstadoChavetas chavetas;

    @Builder.Default
    private Integer chavetasNum = 0;

    private EstadoCompresor compresor;
    private EstadoTanqueAire tanquesDeAire;
    private BigDecimal tiempoDeCargaPsi;
    private BigDecimal tiempoDeCargaTiempo;

    private Dictamen humo;
    private Dictamen gobernado;

    private EstadoFuga cajaDireccion;
    private EstadoFuga depositoAceite;
    private EstadoParabrisas parabrisas;
    private EstadoLimpiaparabrisas limpiaparabrisas;
    private Dictamen huelgo;
    private BigDecimal huelgoCuanto;
    private EstadoEscape escape;

    private String comentarios;

    private List<EvidenciaEvaluacionDTO> evidencias;
}

File: SivemorMobile/app/src/main/java/com/sivemor/mobile/network/dto/EvaluacionRequestDTO.kt
package com.sivemor.mobile.network.dto

data class EvaluacionRequestDTO(
    val id: Long? = null,
    val idVerificacion: Long,
    val idTecnico: Long,

    val lucesGalibo: String? = null,
    val lucesAltas: String? = null,
    val lucesBajas: String? = null,
    val lucesDemarcadorasDelanteras: String? = null,
    val lucesDemarcadorasTraseras: String? = null,
    val lucesIndicadoras: String? = null,
    val faroIzquierdo: String? = null,
    val faroDerecho: String? = null,
    val lucesDireccionalesDelanteras: String? = null,
    val lucesDireccionalesTraseras: String? = null,

    val llantasRinesDelanteros: String? = null,
    val llantasRinesTraseros: String? = null,
    val llantasMasasDelanteras: String? = null,
    val llantasMasasTraseras: String? = null,

    val llantasPresionDelanteraIzquierda: Double? = null,
    val llantasPresionDelanteraDerecha: Double? = null,
    val llantasPresionTraseraIzquierda1: Double? = null,
    val llantasPresionTraseraIzquierda2: Double? = null,
    val llantasPresionTraseraDerecha1: Double? = null,
    val llantasPresionTraseraDerecha2: Double? = null,

    val llantasProfundidadDelanteraIzquierda: Double? = null,
    val llantasProfundidadDelanteraDerecha: Double? = null,
    val llantasProfundidadTraseraIzquierda1: Double? = null,
    val llantasProfundidadTraseraIzquierda2: Double? = null,
    val llantasProfundidadTraseraDerecha1: Double? = null,
    val llantasProfundidadTraseraDerecha2: Double? = null,

    val llantasBirlosDelanteraIzquierda: String? = null,
    val llantasBirlosDelanteraDerecha: String? = null,
    val llantasBirlosTraseraIzquierda: String? = null,
    val llantasBirlosTraseraDerecha: String? = null,

    val llantasBirlosDelanteraIzquierdaNum: Int = 0,
    val llantasBirlosDelanteraDerechaNum: Int = 0,
    val llantasBirlosTraseraIzquierdaNum: Int = 0,
    val llantasBirlosTraseraDerechaNum: Int = 0,

    val llantasTuercasDelanteraIzquierda: String? = null,
    val llantasTuercasDelanteraDerecha: String? = null,
    val llantasTuercasTraseraIzquierda: String? = null,
    val llantasTuercasTraseraDerecha: String? = null,

    val llantasTuercasDelanteraIzquierdaNum: Int = 0,
    val llantasTuercasDelanteraDerechaNum: Int = 0,
    val llantasTuercasTraseraIzquierdaNum: Int = 0,
    val llantasTuercasTraseraDerechaNum: Int = 0,

    val brazoPitman: String? = null,
    val manijasDePuertas: String? = null,
    val chavetas: String? = null,
    val chavetasNum: Int = 0,

    val compresor: String? = null,
    val tanquesDeAire: String? = null,
    val tiempoDeCargaPsi: Double? = null,
    val tiempoDeCargaTiempo: Double? = null,

    val humo: String? = null,
    val gobernado: String? = null,

    val cajaDireccion: String? = null,
    val depositoAceite: String? = null,
    val parabrisas: String? = null,
    val limpiaparabrisas: String? = null,
    val huelgo: String? = null,
    val huelgoCuanto: Double? = null,
    val escape: String? = null,

    val comentarios: String? = null,
    val evidencias: List<EvidenciaEvaluacionDTO> = emptyList()
)